docs/src/main/asciidoc/fault-tolerance-guide.adoc

```diff
@@ -0,0 +1,472 @@
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        try {
+            Integer availability = coffeeRepository.getAvailability(coffee);
+            LOGGER.infof("CoffeeResource#availability() invocation #%d returning successfully", invocationNumber);
+            return Response.ok(availability).build();
+        } catch (RuntimeException e) {
+            String message = e.getClass().getSimpleName() + ": " + e.getMessage();
+            LOGGER.errorf("CoffeeResource#availability() invocation #%d failed: %s", invocationNumber, message);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .type(MediaType.TEXT_PLAIN_TYPE)
+                    .build();
+        }
+    }
+    ...
+}
+----
+
+We added another functionality - the application can return amount of remaining packages of given coffee on our store
+(just a random number).
+
+This time an artificial failure was introduced in the CDI bean: `CoffeeRepositoryService#getAvailability()` method is
+going to alternate between two successful and two failed invocations.
+
+We also added a `@CircuitBreaker` annotation with `requestVolumeThreshold = 4`. `CircuitBreaker.failureRatio` is
+by default 0.5, and `CircuitBreaker.delay` is by default 5 seconds. That means that a circuit breaker will open
+when 2 of the last 4 invocations failed and it will stay open for 5 seconds.
+
+To test this out, do the following:
+
+1. Go to `http://localhost:8080/coffee/2/availability` in your browser. You should see a number being returned.
+2. Hit refresh, this second request should again be successful and return a number.
+3. Refresh two more times. Both times you should see text "RuntimeException: Service failed.", which is the exception
+   thrown by `CoffeeRepositoryService#getAvailability()`.
+4. Refresh couple more times. Unless you waited too long, you should again see exception, but this time it's
+   "CircuitBreakerOpenException: getAvailability". This exception indicates that the circuit breaker opened
+   and the `CoffeeRepositoryService#getAvailability()` method is not being called anymore.
+5. Give it 5 seconds during which circuit breaker should close and you should be able to make two successful requests
+   again.
+
+== Conclusion
+
+MicroProfile Fault Tolerance allows to improve resiliency of your application, without having impact on the complexity
+of our busyness logic.
+
+All that is needed to enable the fault tolerance features in Quarkus is:
+
+* adding `smallrye-fault-tolerance` Quarkus extension to your project using `quarkus-maven-plugin`:
+
+  `mvn quarkus:add-extension -Dextensions="smallrye-fault-tolerance"`
+
+* or simply adding following Maven dependency:
+
+  [source,xml]
+  ----
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-fault-tolerance-deployment</artifactId>
+      <version>${quarkus.version}</version>
```

Comment:
2 things: you need to drop the `-deployment` suffix now (it's a last minute change we made to Quarkus) and you can drop the version, people should use the BOM.